Corona-Warn-App/src/main/java/de/rki/coronawarnapp/nearby/modules/diagnosiskeysdatamapper/DefaultDiagnosisKeysDataMapper.kt

```diff
@@ -0,0 +1,57 @@
+package de.rki.coronawarnapp.nearby.modules.diagnosiskeysdatamapper
+
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.nearby.exposurenotification.DiagnosisKeysDataMapping
+import com.google.android.gms.nearby.exposurenotification.ExposureNotificationClient
+import com.google.android.gms.nearby.exposurenotification.ExposureNotificationStatusCodes.FAILED_RATE_LIMITED
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+@Singleton
+class DefaultDiagnosisKeysDataMapper @Inject constructor(
+    private val client: ExposureNotificationClient,
+    private val appConfigProvider: AppConfigProvider
+) : DiagnosisKeysDataMapper {
+    private suspend fun getDiagnosisKeysDataMapping(): DiagnosisKeysDataMapping? =
+        suspendCoroutine { cont ->
+            client.diagnosisKeysDataMapping
+                .addOnSuccessListener { cont.resume(it) }
+                .addOnFailureListener { cont.resumeWithException(it) }
+        }
+
+    private suspend fun setDiagnosisKeysDataMapping(diagnosisKeysDataMapping: DiagnosisKeysDataMapping) =
+        suspendCoroutine<Unit> { cont ->
+            client.setDiagnosisKeysDataMapping(diagnosisKeysDataMapping)
+                .addOnSuccessListener { cont.resume(Unit) }
+                .addOnFailureListener { cont.resumeWithException(it) }
+        }
+
+    override suspend fun updateDiagnosisKeysDataMapping() {
+        val currentDiagnosisKeysDataMapping = getDiagnosisKeysDataMapping()
+        val newDiagnosisKeysDataMapping = appConfigProvider
```

Comment:
I think it's nicer if we pass `ExposureWindowRiskCalculationConfig` via function parameter. Updating the mapping is always tied to a submission, as the app config can change asynchronously, this could lead to 2 different configs being used in the task and here.

Task gets Appconfig -> ENFClient provide diagnosisKeys(config, files)

Reply:
Invocation has been moved to provide diagnosis key task